iothub/service/src/Registry/Models/BulkRegistryOperationResult.cs

```diff
@@ -1,29 +1,26 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
 using Newtonsoft.Json;
 
 namespace Microsoft.Azure.Devices
 {
     /// <summary>
     /// Encapsulates the result of a bulk registry operation.
     /// </summary>
-    public sealed class BulkRegistryOperationResult
+    public class BulkRegistryOperationResult
     {
         /// <summary>
         /// Whether or not the operation was successful.
         /// </summary>
         [JsonProperty(PropertyName = "isSuccessful", Required = Required.Always)]
-        public bool IsSuccessful { get; internal set; }
+        public bool IsSuccessful { get; protected internal set; }
 
         /// <summary>
-        /// If the operation was not successful, this contains an array of DeviceRegistryOperationError objects.
+        /// If the operation was not successful, this contains DeviceRegistryOperationError objects.
         /// </summary>
         [JsonProperty(PropertyName = "errors", DefaultValueHandling = DefaultValueHandling.Ignore)]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage(
-            "Performance",
-            "CA1819:Properties should not return arrays",
-            Justification = "Cannot change property types on public classes.")]
-        public DeviceRegistryOperationError[] Errors { get; internal set; }
+        public IList<DeviceRegistryOperationError> Errors { get; internal set; } = new List<DeviceRegistryOperationError>();
```

Comment:
Out of curiosity, why do we use `IList` instead of `List`? On another note, if we used `List`, the expression could be simplified with just a `new()` statement, but I assume you had a reason for the alternative instead.

Reply:
Because we prefer to abstract API interfaces away to interfaces instead of concrete classes when possible. We could change the underlying type, as long as it inherits from IList, and not break the customer.